src/main/java/life/qbic/omero/BasicOMEROClient.java

```diff
@@ -71,120 +78,109 @@ private void connect(String hostname, int port, String username, String password
+    return loadAnnotationsForImage(imageID, MapAnnotationData.class);
   }
 
-  private Collection<? extends AnnotationData> loadAnnotationsForImage(long imageID,
-      Class<? extends AnnotationData> type)
-      throws DSOutOfServiceException, DSAccessException, ExecutionException {
+  /**
+   *
+   * @param imageID the omero identifier for the desired image
+   * @param <T> the desired subclass of {@link AnnotationData}
+   * @return a List containing annotation data for the given image matching the desired class
+   */
+  private <T extends AnnotationData> List<T> loadAnnotationsForImage(long imageID, Class<T> type) {
     long userID = this.gateway.getLoggedInUser().getId();
+    ImageData image;
+    List<AnnotationData> annotations;
 
-    BrowseFacility browse = gateway.getFacility(BrowseFacility.class);
-    ImageData image = browse.getImage(this.ctx, imageID);
+    try {
+      BrowseFacility browse = gateway.getFacility(BrowseFacility.class);
+      image = browse.getImage(this.ctx, imageID);
+    } catch (DSOutOfServiceException dsOutOfServiceException) {
+      throw new RuntimeException("Error while accessing omero service: broken connection, expired session or not logged in", dsOutOfServiceException);
+    } catch (ExecutionException executionException) {
+      throw new RuntimeException("Task aborted unexpectedly.", executionException);
+    } catch (DSAccessException dsAccessException) {
+      throw new RuntimeException("Could not pull data from the omero server.", dsAccessException);
+    }
 
-    List<Long> userIds = new ArrayList<Long>();
+    List<Long> userIds = new ArrayList<>();
     userIds.add(userID);
 
-    List<Class<? extends AnnotationData>> types = new ArrayList<Class<? extends AnnotationData>>();
+    List<Class<? extends AnnotationData>> types = new ArrayList<>();
     types.add(type);
 
-    MetadataFacility metadata = gateway.getFacility(MetadataFacility.class);
-    List<AnnotationData> annotations = metadata.getAnnotations(ctx, image, types, userIds);
-
+    try {
+      MetadataFacility metadata = gateway.getFacility(MetadataFacility.class);
+      annotations = metadata.getAnnotations(ctx, image, types, userIds);
+    } catch (DSOutOfServiceException dsOutOfServiceException) {
+      throw new RuntimeException("Error while accessing omero service: broken connection, expired session or not logged in", dsOutOfServiceException);
+    } catch (ExecutionException executionException) {
+      throw new RuntimeException("Task aborted unexpectedly.", executionException);
+    } catch (DSAccessException dsAccessException) {
+      throw new RuntimeException("Could not pull data from the omero server.", dsAccessException);
+    }
     disconnect();
 
-    if (annotations != null) {
-      return annotations;
-    }
-    return new ArrayList<>();
+    return annotations.stream().map(annotationData -> (T) annotationData).collect(Collectors.toList());
```

Comment:
Does this really work if the annotations object is null?

Reply:
afaik it can be null if no annotations exist - which will not throw an error

Reply:
I think annotations is never null since `metadata.getAnnotations` is always executed when we reach the return statement

Reply:
metadata.getAnnotations(ctx, image, types, userIds) can indeed return null, which is not specified in the Omero Doc...weird.

Reply:
urgh... not good.

Reply:
```suggestion
return (annotations != null) ? annotations.stream().map(annotationData -> (T) annotationData).collect(Collectors.toList()) : new ArrayList<T>();
```